app/models/request/sample_compound_aliquot_transfer.rb

```diff
@@ -50,14 +50,40 @@ def aliquots_by_tags_combination
     asset.aliquots.group_by(&:tags_combination)
   end
 
+  # For a group of source aliquots, find or create a compound sample containing the component samples
+  # Assign the compound sample to the target asset
   def transfer_into_compound_sample_aliquot(source_aliquots)
     compound_aliquot = CompoundAliquot.new(request: self, source_aliquots: source_aliquots)
-    unless compound_aliquot.valid?
-      raise Request::SampleCompoundAliquotTransfer::Error, compound_aliquot.errors.full_messages
-    end
 
-    compound_aliquot.create_compound_sample
+    # Check if a compound sample already exists with the source_aliquot samples
+    existing_compound_sample = find_compound_sample_for_components_samples(source_aliquots)
+    if existing_compound_sample
+      compound_aliquot.compound_sample = existing_compound_sample
+    else
+      unless compound_aliquot.valid?
+        raise Request::SampleCompoundAliquotTransfer::Error, compound_aliquot.errors.full_messages
+      end
 
+      compound_aliquot.create_compound_sample
+    end
     target_asset.aliquots.create(compound_aliquot.aliquot_attributes)
   end
+
+  def find_compound_sample_for_components_samples(source_aliquots)
+    # Get all the component samples
+    component_samples = source_aliquots.map(&:sample)
+
+    # Get all the compound samples the first component sample
+    compound_samples = component_samples[0].compound_samples
+
+    # If there is only 1 compound sample, return it. Otherwise, return the last created compound sample
+    # NPG have confirmed we do not need to fix the data where there are multiple compound samples with
+    # the same component samples
```

Comment:
I would move this comment from `# NPG have...` onwards to under the `# Due to previous impl..` one below, because I think it makes more sense after reading that.